src/admin/resolvers/mutations/ApprovedItem.auth.integration.ts

```diff
@@ -0,0 +1,592 @@
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: UPDATE_APPROVED_ITEM,
+        variables: { data: input },
+      });
+
+      expect(result.data).to.be.null;
+      expect(result.errors).not.to.be.null;
+      expect(result.errors?.[0].message).to.contain(ACCESS_DENIED_ERROR);
+
+      await server.stop();
+    });
+  });
+
+  describe('updateApprovedCorpusItemAuthors mutation', () => {
+    let item: ApprovedItem;
+    let authors: ApprovedItemAuthor[];
+    let input: UpdateApprovedItemAuthorsInput;
+
+    beforeEach(async () => {
+      item = await createApprovedItemHelper(db, {
+        title: "3 Things Everyone Knows About LEGO That You Don't",
+        status: CuratedStatus.RECOMMENDATION,
+        language: 'EN',
+      });
+
+      authors = [
+        { name: 'Author One', sortOrder: 1 },
+        { name: 'Author Two', sortOrder: 2 },
+      ];
+
+      input = {
+        externalId: item.externalId,
+        authors,
+      };
+    });
+
+    it('should succeed if user has access to one of scheduled surfaces', async () => {
+      // Set up auth headers with access to a single Scheduled Surface
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: `group1,group2,${MozillaAccessGroup.NEW_TAB_CURATOR_ENGB}`,
+      };
+
+      const server = getServerWithMockedHeaders(headers, eventEmitter);
+      await server.start();
+
+      // Set up event tracking
+      const eventTracker = sinon.fake();
+      eventEmitter.on(ReviewedCorpusItemEventType.UPDATE_ITEM, eventTracker);
+
+      const res = await server.executeOperation({
+        query: UPDATE_APPROVED_ITEM_AUTHORS,
+        variables: { data: input },
+      });
+
+      // Good to check for any errors before proceeding with the rest of the test
+      expect(res.errors).not.to.exist;
+      const data = res.data;
+
+      // External ID should be unchanged
+      expect(data?.updateApprovedCorpusItemAuthors.externalId).to.equal(
+        item.externalId
+      );
+
+      // Updated properties should be... updated
+      expect(data?.updateApprovedCorpusItemAuthors.authors).to.deep.equal(
+        input.authors
+      );
+
+      // The `updatedBy` field should now be the SSO username of the user
+      // who updated this record
+      expect(data?.updateApprovedCorpusItemAuthors.updatedBy).to.equal(
+        headers.username
+      );
+
+      // Check that the UPDATE_ITEM event was fired successfully:
+      // 1 - Event was fired once!
+      expect(eventTracker.callCount).to.equal(1);
+      // 2 - Event has the right type.
+      expect(await eventTracker.getCall(0).args[0].eventType).to.equal(
+        ReviewedCorpusItemEventType.UPDATE_ITEM
+      );
+      // 3- Event has the right entity passed to it.
+      expect(
+        await eventTracker.getCall(0).args[0].reviewedCorpusItem.externalId
+      ).to.equal(data?.updateApprovedCorpusItemAuthors.externalId);
```

Comment:
nit - can we keep the auth tests focused on auth stuff only? we are testing the data in the non-auth tests already, no need to duplicate here.

Reply:
Done!